packages/adminrouter/extra/src/test-harness/tests/test_cache.py

```diff
@@ -871,6 +871,34 @@ def test_ip_per_task_app_without_user_networking(
             req_data = resp.json()
             assert req_data['endpoint_id'] == 'http://127.0.0.2:80'
 
+    def test_ip_per_task_app_with_unspecified_ip_address_DCOS_16592(
+            self, nginx_class, mocker, valid_user_header):
+        """
+        Test that an app that, instead of specifying 'ipAddress: null' does not
+        specify 'ipAddress' at all, is successfully cached.
+        """
+        app = self._scheduler_alwaysthere_app()
+
+        # Remove the 'ipAddress' key completely, thereby triggering DCOS-16592.
+        del(app["ipAddress"])
+
+        ar = nginx_class()
+
+        mocker.send_command(endpoint_id='http://127.0.0.1:8080',
+                            func_name='set_apps_response',
+                            aux_data={"apps": [app]})
+
+        url = ar.make_url_from_path('/service/scheduler-alwaysthere/foo/bar/')
+        with GuardedSubprocess(ar):
+            # Trigger cache update by issuing request:
+            resp = requests.get(url,
+                                allow_redirects=False,
+                                headers=valid_user_header)
+
+            assert resp.status_code == 200
```

Comment:
This was copy-pasted from the test right above it - in the spirit of uniformity i'm going to leave this as is.

https://github.com/dcos/dcos/pull/1708/files/f07b0ccd2e97cd3612b10517a93747898a1dba03#diff-dfe9720a19c629454342fef91b250626R870